frappe/installer.py

```diff
@@ -43,6 +43,8 @@ def install_db(root_login="root", root_password=None, db_name=None, source_sql=N
 	frappe.connect(db_name=db_name)
 	check_if_ready_for_barracuda()
 	import_db_from_sql(source_sql, verbose)
+	# Long imports cause the connection to time out on the server, so close and reconnect.
+	frappe.db.close()
```

Comment:
This doesn't create a new connection, just destroys the Python client's object. 

Does this work for you? If yes, do you have a clue as to how/when a new connection is created? and Can we use connect instead (like the PR linked in the previous comment), since it's more explicit?

Reply:
it works probably because of implicit reconnect in `db.sql`

https://github.com/frappe/frappe/blob/28e57d9d3d89ee0fab8e00a5075ca89f1b5f50c5/frappe/database/database.py#L126-L127